tokio-util/src/net/mod.rs

```diff
@@ -0,0 +1,104 @@
+//! TCP/UDP/Unix helpers for tokio.
+
+use crate::either::Either;
+use std::future::Future;
+use std::io::Result;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+#[cfg(unix)]
+pub mod unix;
+
+/// A trait for a listener: `TcpListener` and `UnixListener`.
+pub trait Listener: Send + Unpin {
+    /// The stream's type of this listener.
+    type Io: tokio::io::AsyncRead + tokio::io::AsyncWrite;
+    /// The socket address type of this listener.
+    type Addr;
+
+    /// Polls to accept a new incoming connection to this listener.
+    fn poll_accept(&self, cx: &mut Context<'_>) -> Poll<Result<(Self::Io, Self::Addr)>>;
```

Comment:
This should take a pinned reference to self.

Reply:
I replicated the signature of the function from tokio's TCP listener. Are you sure you want to make it different?

Reply:
In general, making this take `&self` will make it difficult to implement this trait for anything else than the two listener types in Tokio proper. This is why I suggested changing it.

Reply:
Ok I see

Reply:
Fixed in fe231ec6